### ISSUE TITLE
from __future__ import annotations ignored

### ISSUE THREAD

**Issue by clo-vis** [PyCQA/pylint#6523]
### Bug description

```python
from __future__ import annotations
from csv import DictWriter
from typing import IO


def f_is_for_fun(file: IO[str]) -> DictWriter[str]:
    return DictWriter(file, ["id", "name"])
```

Reference https://github.com/PyCQA/pylint/issues/6519

### Command used

```shell
pylint playground.py
```

### Pylint output

```shell
************* Module playground
…
playground.py:6:35: E1136: Value 'DictWriter' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior

No error, as the python interpreter doesn't complain, nor does mypy, and https://github.com/PyCQA/pylint/issues/3320 is closed.

### Pylint version

```shell
pylint 2.13.8
astroid 2.11.4
Python 3.9.12 (tags/v3.9.12:b28265d, Mar 23 2022, 23:52:46) [MSC v.1929 64 bit (AMD64)]
```

**Comment by hmc-cs-mdrissi** [comment]
More broadly generic type may not be from typeshed but from other .pyi files which may be defined in arbitrary locations (well depends on type checker's search paths for stubs). As an example mypy supports MYPY_PATH allowing defining other places for type stubs separate from python path. pyright similarly has it's own setting for possible places .pyi files can be placed. I use custom type stubs for 3rd party libraries that lack type hints.

**Comment by cdce8p** [comment]
@hmc-cs-mdrissi You're right of course. Pylint isn't a type checker and thus doesn't use `.pyi` files at all.
Probably the best way forward for us is thus to ignore all `unsubscriptable-object` errors in string annotations and with `from __future__ import annotations`. Those don't result in runtime errors and a type checker would catch them anyway.

I've opened #6536 to make that change.
